api/capi/include/nntrainer_internal.h

```diff
@@ -101,7 +101,7 @@ typedef struct {
  */
 typedef struct {
   uint magic;
-  std::shared_ptr<ml::train::Dataset> dataset;
+  std::array<std::shared_ptr<ml::train::Dataset>, 3> dataset;
```

Comment:
In the capi point of view, the change will be abstracted away by dataset having 3 databuffers.